src/Stratis.Features.FederatedPeg/Collateral/CollateralChecker.cs

```diff
@@ -211,7 +233,28 @@ public bool CheckCollateral(IFederationMember federationMember)
 
             lock (this.locker)
             {
-                return (this.depositsByAddress[member.CollateralMainchainAddress] ?? 0) >= member.CollateralAmount;
+                AddressIndexerData balanceData = this.balancesDataByAddress[member.CollateralMainchainAddress];
+
+                if (balanceData == null)
+                {
+                    // No data. Assume collateral is 0. It's ok if there is no collateral set for that fed member.
+                    this.logger.LogTrace("(-)[NO_DATA]");
+                    return 0 >= member.CollateralAmount;
+                }
+
+                long balance = 0;
+
+                foreach (AddressBalanceChange change in balanceData.BalanceChanges.Where(x => x.BalanceChangedHeight <= heightToCheckAt))
```

Comment:
I've seen this bit of code somewhere else... can we maybe put this "compacting" bit of code into a method or extension or something?

Reply:
Fixed